src/cli/managers/roundset_manager.py

```diff
@@ -1,19 +1,191 @@
+import os
 from pathlib import Path
-from typing import Tuple
+import shutil
+import subprocess
+from subprocess import CalledProcessError, TimeoutExpired
+from typing import Any, cast, Dict, Tuple
+
+from cli.output_converter import OutputConverter
+from cli.generators.network_generator import FullyConnectedNetworkGenerator
+from cli.generators.result_generator import ErrorResultGenerator
+from cli.generators.template_generator import RoleTemplate, RoleMappingTemplate, NetworkTemplate
+from cli.type_aliases import AssetType, GeneratedResultType
 
 
 class RoundSetManager:
-    def __init__(self) -> None:
-        pass
+    def __init__(self, asset: Dict[str, Any], path: Path) -> None:
+        self.__asset = asset
+        self.__path = path
+        self.__input_dir = str(path / "input")
+        self.__log_dir = str(path / "raw_output")
+        self.__output_dir = "LAST"
+        self.__fully_connected_network_generator = FullyConnectedNetworkGenerator()
+        self.__output_converter = OutputConverter(
+            log_dir=self.__log_dir,
+            output_dir=self.__output_dir,
+            instruction_converter=self.__fully_connected_network_generator
+        )
+
+    def process(self) -> GeneratedResultType:
+        round_failed = False
+        round_number = 1
+        self.__output_converter.prepare_output()
+
+        try:
+            self._run_application()
+        except CalledProcessError as exc:
+            exception_type = type(exc).__name__
+            message = f"NetQASM returned with exit status {exc.returncode}."
+            trace = exc.stderr.decode() if exc.stderr is not None else None
+            round_failed = True
+        except TimeoutExpired as exc:
+            exception_type = type(exc).__name__
+            message = f"Call to simulator timed out after {exc.timeout} seconds."
+            trace = exc.stderr.decode() if exc.stderr is not None else None
+            round_failed = True
+        except Exception as exc:
+            exception_type = type(exc).__name__
+            trace = None
+            message = str(exc)
+            round_failed = True
 
-    def process(self) -> None:
-        pass
+        if round_failed:
+            result = ErrorResultGenerator.generate(round_number, exception_type, message, trace)
+        else:
+            result = self.__output_converter.convert(round_number)
+
+        return result
 
     def validate_asset(self, path: Path) -> Tuple[bool, str]:
```

Comment:
Isn't this validate_asset needed?

Reply:
I think the 'experiment validate' command already validates the asset. So the experiment run command does not need validate_asset